src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/PayloadPreparationService.cs

```diff
@@ -166,10 +166,7 @@ private void CleanupOldPayloads(object? sender, EventArgs e)
         {
             if (_payloadStorage.TryGetValue(payloadId, out IBlockImprovementContext? blockContext))
             {
-                using (blockContext)
-                {
-                    return blockContext.CurrentBestBlock;
-                }
+                return blockContext.CurrentBestBlock;
```

Comment:
This is actually a downgrade. I think we want to stop improvements on first get.